app/models/setting.rb

```diff
@@ -9,13 +9,22 @@ class Setting < ActiveRecord::Base
   NONZERO_ATTRS = %w{ puppet_interval idle_timeout entries_per_page max_trend outofsync_interval }
   BLANK_ATTRS = %w{ trusted_puppetmaster_hosts login_delegation_logout_url authorize_login_delegation_auth_source_user_autocreate root_pass default_location default_organization websockets_ssl_key websockets_ssl_cert oauth_consumer_key oauth_consumer_secret }
   URI_ATTRS = %w{ foreman_url unattended_url }
+  URI_BLANK_ATTRS = %w{ login_delegation_logout_url } + URI_ATTRS
+  IP_ATTRS = %w{ libvirt_default_console_address }
+  IP_REGEXP_ATTRS = %w{ remote_addr }
 
   class UriValidator < ActiveModel::EachValidator
     def validate_each(record, attribute, value)
-      record.errors.add attribute, _("must be a valid URI") unless %w(http https).include? URI(value).scheme
```

Comment:
I missed this one, but this code can be removed, and use `app/validators/url_schema_validator.rb`